src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java

```diff
@@ -143,63 +180,40 @@ public void beginTree(DetailAST rootAST) {
-            case TokenTypes.MINUS_ASSIGN:
-            case TokenTypes.STAR_ASSIGN:
-            case TokenTypes.DIV_ASSIGN:
-            case TokenTypes.MOD_ASSIGN:
-            case TokenTypes.SR_ASSIGN:
-            case TokenTypes.BSR_ASSIGN:
-            case TokenTypes.SL_ASSIGN:
-            case TokenTypes.BAND_ASSIGN:
-            case TokenTypes.BXOR_ASSIGN:
-            case TokenTypes.BOR_ASSIGN:
-                visitAssign(ast);
-                break;
-            case TokenTypes.INC:
-            case TokenTypes.POST_INC:
-            case TokenTypes.DEC:
-            case TokenTypes.POST_DEC:
-                visitIncDec(ast);
-                break;
-            default:
-                throw new IllegalStateException(ast.toString());
+        final int type = ast.getType();
+        if (TokenUtil.isOfType(type, TokenTypes.CTOR_DEF, TokenTypes.METHOD_DEF)) {
+            visitMethodDef(ast);
+        }
+        else if (type == TokenTypes.LAMBDA) {
+            if (ast.getParent().getType() != TokenTypes.SWITCH_RULE) {
+                visitLambda(ast);
+            }
+        }
+        else if (TokenUtil.isOfType(type, ASSIGN_TOKENS)) {
+            visitAssign(ast);
+        }
+        else if (TokenUtil.isOfType(type, INC_DEC_TOKENS)) {
+            visitIncDec(ast);
+        }
+        else {
+            throw new IllegalStateException(ast.toString());
         }
     }
 
     @Override
     public void leaveToken(DetailAST ast) {
-        switch (ast.getType()) {
-            case TokenTypes.CTOR_DEF:
-            case TokenTypes.METHOD_DEF:
-                leaveMethodDef();
-                break;
-            case TokenTypes.ASSIGN:
-            case TokenTypes.PLUS_ASSIGN:
-            case TokenTypes.MINUS_ASSIGN:
-            case TokenTypes.STAR_ASSIGN:
-            case TokenTypes.DIV_ASSIGN:
-            case TokenTypes.MOD_ASSIGN:
-            case TokenTypes.SR_ASSIGN:
-            case TokenTypes.BSR_ASSIGN:
-            case TokenTypes.SL_ASSIGN:
-            case TokenTypes.BAND_ASSIGN:
-            case TokenTypes.BXOR_ASSIGN:
-            case TokenTypes.BOR_ASSIGN:
-            case TokenTypes.INC:
-            case TokenTypes.POST_INC:
-            case TokenTypes.DEC:
-            case TokenTypes.POST_DEC:
-                // Do nothing
-                break;
-            default:
-                throw new IllegalStateException(ast.toString());
+        final int type = ast.getType();
+        if (TokenUtil.isOfType(type, TokenTypes.CTOR_DEF, TokenTypes.METHOD_DEF)) {
+            popParameterNamesStack();
+        }
+        else if (type == TokenTypes.LAMBDA) {
+            if (ast.getParent().getType() != TokenTypes.SWITCH_RULE) {
+                popParameterNamesStack();
+            }
+        }
+        else if (!TokenUtil.isOfType(type, ASSIGN_TOKENS)
+                && !TokenUtil.isOfType(type, INC_DEC_TOKENS)) {
+            throw new IllegalStateException(ast.toString());
```

Comment:
I dont think you need this case. In switch it was there because there was violation about absense of default case, therefore there was nop for these token types and exception for default. In if-else you do not need that.

Reply:
Done, removed the related test also.